src/Cpu.re

```diff
@@ -43,19 +55,56 @@ let debug_log = cpu => {
   );
 };
 
-exception InstructionNotImplemented(int);
+let jump = (cpu, argument) => {
+  cpu.pc = argument;
+};
+
+let get_argument = (cpu: t, mode: Opcode.addressing_mode) => {
+  switch (mode) {
+  | Absolute => Memory.get_word(cpu.memory, cpu.pc)
+  | _ => raise(AddressingModeNotImplemented(mode))
+  };
+};
 
-let jump = cpu => {
-  cpu.pc = Memory.get_word(cpu.memory, cpu.pc);
-  cpu.cycles = cpu.cycles + 3;
+let handle = (definition: Instruction.t, opcode: Opcode.t, cpu: t) => {
+  let argument = get_argument(cpu, opcode.addressing_mode);
+
+  let operation =
+    switch (definition.label) {
+    | "jmp" => jump
+    | _ => raise(InstructionNotImplemented(definition.label))
+    };
+
+  operation(cpu, argument);
+
+  cpu.cycles = cpu.cycles + opcode.timing;
 };
 
+let add_instructions = (definition: Instruction.t) =>
```

Comment:
Seems like this should be named `add_instruction` since it's just handling all the opcode variants for a _single_ instruction, no?